app/models/duffel.rb

```diff
@@ -7,6 +7,39 @@ def initialize(contents)
 
   def add_item(item_id)
     contents[item_id.to_s] ||= 0
-    contents[item_id.to_s] += 0
+    contents[item_id.to_s] += 1
+  end
+
+  def total
+    contents.values.sum
+  end
+
+  def count_of(item_id)
+    contents[item_id.to_s]
+  end
+
+  def subtotal
+    item_prices = contents.map do |item_id, quantity|
+      item = Item.find(item_id)
+      item.price * quantity
+    end
+    item_prices.sum
+  end
+
+  def duffel_items
+    contents.map do |item_id, _quantity|
+      Item.find(item_id)
+    end
+  end
+
+  def item_details
```

Comment:
I've now got the `item_subtotal` method included in the model. I decided to leave the `item_details` method, just in case we see a reason for it later. It can easily be removed if we want to.